Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/transform/CoreX12AuditDataTransform.java

```diff
@@ -0,0 +1,467 @@
+            } else if (msg instanceof COREEnvelopeBatchSubmission) {
+                payloadId = ((COREEnvelopeBatchSubmission) msg).getPayloadID();
+            } else if (msg instanceof COREEnvelopeBatchSubmissionResponse) {
+                payloadId = ((COREEnvelopeBatchSubmissionResponse) msg).getPayloadID();
+            }
+            participantObject.setParticipantObjectID(payloadId);
+        } else {
+            return participantObject;
+        }
+        byteArray = marshallToByteArray(msg);
+        TypeValuePairType oType = new TypeValuePairType();
+        oType.setValue(byteArray);
+        participantObject.getParticipantObjectDetail().add(oType);
+
+        // Set the Participation Object Typecode
+        participantObject.setParticipantObjectTypeCode(AuditDataTransformConstants.PARTICIPANT_OJB_TYPE_CODE_SYSTEM);
+
+        // Set the Participation Object Typecode Role
+        participantObject.setParticipantObjectTypeCodeRole(CoreX12AuditDataTransformConstants.PARTICIPANT_OJB_TYPE_CODE_ROLE_X12);
+
+        // Set the Participation Object Id Type code
+        CodedValueType partObjIdTypeCode = new CodedValueType();
+        partObjIdTypeCode.setCode(payloadId);
+        partObjIdTypeCode.setCodeSystemName(CoreX12AuditDataTransformConstants.CAQH_x12_CONNECTIVITY_CODED_SYS_NAME);
+        partObjIdTypeCode.setDisplayName(CoreX12AuditDataTransformConstants.CAQH_x12_CONNECTIVITY_CODED_SYS_DISPLAY_NAME);
+        participantObject.setParticipantObjectIDTypeCode(partObjIdTypeCode);
+
+        return participantObject;
+    }
+
+    private static byte[] marshallToByteArray(Object msg) {
+        byte[] bObject = null;
+        COREEnvelopeRealTimeRequest oRequestNoPayload = null;
+        COREEnvelopeRealTimeResponse oResponseNoPayload = null;
+        COREEnvelopeBatchSubmission oBatchRequestNoPayload = null;
+        COREEnvelopeBatchSubmissionResponse oBatchResponseNoPayload = null;
+        javax.xml.namespace.QName xmlqname = null;
+        JAXBContextHandler oHandler = new JAXBContextHandler();
+        try {
+            JAXBContext jc = oHandler.getJAXBContext(CoreX12AuditDataTransformConstants.CORE_X12_JAXB_CONTEXT);
+            Marshaller marshaller = jc.createMarshaller();
+            ByteArrayOutputStream baOutStrm = new ByteArrayOutputStream();
+            if (msg instanceof COREEnvelopeRealTimeRequest) {
+                oRequestNoPayload = (COREEnvelopeRealTimeRequest) msg;
+                oRequestNoPayload.setPayload("");
+                xmlqname = new javax.xml.namespace.QName(CoreX12AuditDataTransformConstants.CORE_X12_NAMESPACE_URI, CoreX12AuditDataTransformConstants.CORE_X12_REQUEST_LOCALPART);
+                JAXBElement<org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeRequest> element = new JAXBElement<org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeRequest>(xmlqname, COREEnvelopeRealTimeRequest.class, oRequestNoPayload);
+                marshaller.marshal(element, baOutStrm);
```

Comment:
these two bottom lines can be pulled from each if statement.  repetitive functionality.